email: Update realm deactivation email for data deletion information.

laurynmm opened the pull request:
Fixes #24686.

**Notes**:
- If there is a scheduled data deletion in the future, then the date is localized for the owner's language.
- In the case where no data deletion was scheduled when the realm was deactivated (e.g., via the support view or management command), then there is no change to the email content.

---

**Screenshots and screen captures:**

<details>
<summary>No data deletion done or scheduled when organization was deactivated</summary>

**No change from current email that's sent to organization owners.**
![Screenshot from 2024-12-19 14-08-11](https://github.com/user-attachments/assets/5bcb283d-48d6-4a97-ac55-d6d80ef36d6b)
![Screenshot from 2024-12-19 14-07-59](https://github.com/user-attachments/assets/d24734fa-94e5-4b36-91f5-9d31333e937a)
</details>
<details>
<summary>Data was immediately deleted when organization was deactivated</summary>

![Screenshot from 2024-12-19 14-07-18](https://github.com/user-attachments/assets/77f32682-3b21-4c9c-ab9e-ab2d6126c404)
![Screenshot from 2024-12-19 14-07-08](https://github.com/user-attachments/assets/9f1387de-fdb2-4d56-86e5-6f99da88d557)
</details>
<details>
<summary>Data is scheduled to be deleted in the future when organization was deactivated</summary>

**3 owners - Russian, Italian and English set for their default languages**

![Screenshot from 2024-12-19 14-05-03](https://github.com/user-attachments/assets/f11e1512-6209-4e91-b753-56153abb64ca)
![Screenshot from 2024-12-19 14-04-15](https://github.com/user-attachments/assets/116d3434-ac3d-4642-a96a-5f4469138643)
</details>

---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
It looks funny to have 3 lines here. Let's just put the data deletion line in the first paragraph.

laurynmm commented:
**Updates**:
- Fixed issue with backend test on email with immediate data deletion, which had console output.
- Updated emails to have the data deletion sentence in the first paragraph. See updated screenshots below.

<details>
<summary>Data deletion scheduled</summary>

![Screenshot from 2024-12-19 18-27-42](https://github.com/user-attachments/assets/150a5ba2-89b1-4afc-9ef8-a61afc7a10fa)
![Screenshot from 2024-12-19 18-27-11](https://github.com/user-attachments/assets/671b182f-d3d2-402c-9d68-c549896d3579)
</details>
<details>
<summary>Immediate data deletion</summary>

![Screenshot from 2024-12-19 18-27-27](https://github.com/user-attachments/assets/3587a6c7-1f15-4e11-99c0-1a3096900532)
![Screenshot from 2024-12-19 18-26-56](https://github.com/user-attachments/assets/6350fa37-5895-4a9d-a013-2a250b801f65)
</details>

alya commented:
Do we put line breaks between paragraphs in text-only emails? I feel like that would look better, but I'm OK with not doing it if that's more consistent.

The html emails look good to me!

laurynmm commented:
**Updates**:
- Added another line break to the text version of the email.

<details>
<summary>Updated text version screenshot</summary>

![Screenshot from 2025-01-07 17-21-45](https://github.com/user-attachments/assets/2e5a3c1c-4ab5-4f6b-9e23-854a4c00deca)

</details>

timabbott commented:
Merged, thanks @laurynmm!
